src/v/coproc/ntp_context.h

```diff
@@ -42,11 +43,11 @@ struct ntp_context {
     /// Reference to the storage layer for reading from the input ntp
     storage::log log;
     /// Interested scripts write their last read offset of the input ntp
-    absl::btree_map<script_id, offset_pair> offsets;
+    offsets_map offsets;
 };
 
 using ntp_context_cache
-  = absl::node_hash_map<model::ntp, ss::lw_shared_ptr<ntp_context>>;
+  = absl::flat_hash_map<model::ntp, ss::lw_shared_ptr<ntp_context>>;
```

Comment:
why node -> flat? flat has a bunch of memory constraints. I'm ok w/ it. but we should have a comment about it.